src/xai_grok/errors/unauthorized_user_error.py

```diff
@@ -0,0 +1,8 @@
+import abc
+
+
+class UnauthorizedUserError(Exception, abc.ABC):
+
+    @abc.abstractmethod
+    def __init__(self, message: str):
+        super().__init__(message)
```

Comment:
The `UnauthorizedUserError` class has an abstract `__init__` method which prevents direct instantiation of subclasses without overriding it, but this is unnecessary since `__init__` is not typically made abstract.